ddsrouter_yaml/test/unittest/configuration/YamlReaderConfigurationTest.cpp

```diff
@@ -368,6 +368,117 @@ TEST(YamlReaderConfigurationTest, invalid_routes)
+            R"(
+        version: v3.0
+        participants:
+          - name: "P1"
+            kind: "echo"
+          - name: "P2"
+            kind: "echo"
+        )";
+    Yaml yml = YAML::Load(yml_configuration);
+
+    std::vector<unsigned int> test_cases = {0, 10, 100, 1000, 5000, 10000};
+
+    for (unsigned int test_case : test_cases)
+    {
+        Yaml yml_specs;
+        yml_specs[ddspipe::yaml::MAX_TX_RATE_TAG] = test_case;
+        yml[ddspipe::yaml::SPECS_TAG] = yml_specs;
+
+        // Load configuration
+        ddsrouter::core::DdsRouterConfiguration configuration_result =
+                ddsrouter::yaml::YamlReaderConfiguration::load_ddsrouter_configuration(yml);
+
+        // Check max history depth is correct
+        ASSERT_EQ(test_case, configuration_result.advanced_options.max_tx_rate);
+    }
+}
+
+/**
+ * Test load of max reception rate in the configuration
+ *
+ * CASES:
+ * - trivial configuration
+ */
+TEST(YamlReaderConfigurationTest, max_rx_rate)
+{
+    const char* yml_configuration =
+            // trivial configuration
+            R"(
+        version: v3.0
+        participants:
+          - name: "P1"
+            kind: "echo"
+          - name: "P2"
+            kind: "echo"
+        )";
+    Yaml yml = YAML::Load(yml_configuration);
+
+    std::vector<unsigned int> test_cases = {0, 10, 100, 1000, 5000, 10000};
+
+    for (unsigned int test_case : test_cases)
+    {
+        Yaml yml_specs;
+        yml_specs[ddspipe::yaml::MAX_RX_RATE_TAG] = test_case;
+        yml[ddspipe::yaml::SPECS_TAG] = yml_specs;
+
+        // Load configuration
+        ddsrouter::core::DdsRouterConfiguration configuration_result =
+                ddsrouter::yaml::YamlReaderConfiguration::load_ddsrouter_configuration(yml);
+
+        // Check max history depth is correct
+        ASSERT_EQ(test_case, configuration_result.advanced_options.max_rx_rate);
+    }
+}
+
+/**
+ * Test load of downsampling in the configuration
+ *
+ * CASES:
+ * - trivial configuration
+ */
+TEST(YamlReaderConfigurationTest, downsampling)
+{
+    const char* yml_configuration =
+            // trivial configuration
+            R"(
+        version: v3.0
+        participants:
+          - name: "P1"
+            kind: "echo"
+          - name: "P2"
+            kind: "echo"
+        )";
+    Yaml yml = YAML::Load(yml_configuration);
+
+    std::vector<unsigned int> test_cases = {1, 10, 100, 1000, 5000, 10000};
+
+    for (unsigned int test_case : test_cases)
+    {
+        Yaml yml_specs;
+        yml_specs[ddspipe::yaml::DOWNSAMPLING_TAG] = test_case;
+        yml[ddspipe::yaml::SPECS_TAG] = yml_specs;
+
+        // Load configuration
+        ddsrouter::core::DdsRouterConfiguration configuration_result =
+                ddsrouter::yaml::YamlReaderConfiguration::load_ddsrouter_configuration(yml);
+
+        // Check max history depth is correct
+        ASSERT_EQ(test_case, configuration_result.advanced_options.downsampling);
+    }
+}
```

Comment:
Design same kind of tests but instead of setting these qos in specs, set them in participants section.

Reply:
In these tests we are checking that the DdsRouterConfiguration is loaded correctly; we don't have the mechanisms to ensure that the ParticipantConfiguration is loaded correctly. There are docker tests (https://github.com/eProsima/DDS-Router/tree/feature/manual_topics_conf/ddsrouter_test/compose/test_cases/frequency/downsampling) that only set the downsampling for a Participant and make sure it is behaving accordingly.